pkg/xen-tools/initrd/chroot2.c

```diff
@@ -37,16 +43,17 @@ int main(int argc, char **argv) {
     setsid();
     ioctl(0, TIOCSCTTY, 1);
 
-    chroot(argv[1]);
-    chdir(argv[2]);
-
     uid = strtol(argv[3], &endptr, 10);
     gid = strtol(argv[4], &endptr, 10);
 
-    args.uid = uid;
-    args.gid = gid;
-    args.command = argv[5];
-    args.args = argv + 5;
+    args = (struct clone_args) {
+        .chroot = argv[1],
+        .workdir = argv[2],
+        .uid = uid,
+        .gid = gid,
+        .command = argv[5],
+        .args = argv + 5,
+    };
 
     child_pid = clone(childFunc, child_stack + STACK_SIZE, CLONE_NEWPID | SIGCHLD, (void *)(&args));
```

Comment:
What is the reason to use clone() here? Which data we are sharing with the child process? If we don't need to share memory, than the right syscall to use is fork()....

Reply:
Got it @OhmSpectator , I know he didn't change this line, but I think it was still valid to ask.... all clear now...